internal/provider/metric_export_prometheus_config_resource.go

```diff
@@ -0,0 +1,405 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"github.com/cockroachdb/cockroach-cloud-sdk-go/pkg/client"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/booldefault"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+	"net/http"
+	"strings"
+)
+
+var metricExportPrometheusConfigAttributes = map[string]schema.Attribute{
+	"id": schema.StringAttribute{
+		Required:            true,
+		MarkdownDescription: "Cluster ID.",
+		PlanModifiers: []planmodifier.String{
+			stringplanmodifier.RequiresReplace(),
+		},
+	},
+	"enabled": schema.BoolAttribute{
+		Default:     booldefault.StaticBool(true),
+		Optional:    true,
+		Computed:    true,
+		Description: "Represents prometheus integration status. Please refrain setting any value as it is used for state management.",
+	},
+	"status": schema.StringAttribute{
+		Computed:    true,
+		Description: "Encodes the possible states that a metric export configuration can be in as it is created, deployed, and disabled.",
+	},
+	"user_message": schema.StringAttribute{
+		Computed:    true,
+		Description: "Elaborates on the metric export status and hints at how to fix issues that may have occurred during asynchronous operations.",
+	},
+	"targets": schema.MapAttribute{
+		Computed:    true,
+		ElementType: types.StringType,
+		Description: "Represents the endpoint details to scrape metrics.",
```

Comment:
This description seems a bit vague to me. Actually, looking at the api documentation, I'm still unclear what this is.  In the api we have this:

https://github.com/cockroachlabs/managed-service/blob/41b3b2a651844d2445e2ed914c6ae0ee1e708dff/console/apipb/api.proto#L1604-L1606

>   // targets is a map of ports exposing metrics to regions.
  map<string, string> targets = 4;

So based on the types, it seems like we might have a list of host:port -> region mappings but I'm not sure. @abarganier would you be a good resource to ask about this?  Is there documentation on this somewhere?  Ideally we can improve the api docs here with something clearer.  An example would help a lot.

Reply:
targets would contain the region -> scrape endpoint mapping like below:
`"targets": {
        "australia-southeast2": "http://localhost:7900/api/v1/clusters/c33bdfbe-b107-4d59-b110-ed435ace547a/metricexport/prometheus/australia-southeast2/scrape"
    }`
I have updated the description to improve readability and it's usage. I hope it is clear now.

Reply:
@aa-joshi has it right - this is returned to tell the user which scrape URLs are exposed per-region. These are the URLs that users plug into their Prometheus scrape configs to start scraping metrics.

Reply:
cool.  I'll put up a PR to get this example into the ccapi docs.  I like the new description.  I think the more typical way to access the targets would be to create an Output Value for them. Although I'm unclear on the ramifications of this being a map.  I found this: https://stackoverflow.com/a/55006832